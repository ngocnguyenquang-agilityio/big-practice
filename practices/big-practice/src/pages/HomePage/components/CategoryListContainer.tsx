// Libs
import { useParams } from 'react-router-dom';

// Components
import Collection from '@components/Collection';

const CategoryListContainer = ({ categories }: { categories: Array<{ label: string; value: string }> }) => {
  const { category } = useParams();
  return (
    <Collection
      categories={categories}
      selectingItem={category}
    />
  );
};

export default CategoryListContainer;
