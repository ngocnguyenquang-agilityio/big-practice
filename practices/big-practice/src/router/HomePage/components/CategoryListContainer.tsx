// Libs
import { memo } from 'react';
import { Link, useParams } from 'react-router-dom';

// Helpers
import { isEmpty } from '@helpers/utils';

export const Categories = memo(
  ({
    categories = [],
    selectingItem,
  }: {
    categories: Array<{ value: string; label: string }>;
    selectingItem?: string;
  }): JSX.Element => {
    return (
      <nav>
        <h3 className='text-xs text-neutral-400 md:block'>Category</h3>
        <ul
          className='block'
          data-testid='categories'
        >
          {categories.map((item) => (
            <li
              className={`mt-2 flex text-white ${selectingItem === item.value ? 'font-bold underline' : ''}`}
              key={item.value}
            >
              <Link
                to={isEmpty(item.value) ? '/' : `/category/${item.value}`}
                className='w-full text-sm underline-offset-4 hover:underline hover:text-neutral-100'
              >
                {item.label}
              </Link>
            </li>
          ))}
        </ul>
      </nav>
    );
  },
);

const CategoryListContainer = ({ categories }: { categories: Array<{ label: string; value: string }> }) => {
  const { category } = useParams();
  return (
    <Categories
      categories={categories}
      selectingItem={category}
    />
  );
};

export default CategoryListContainer;
