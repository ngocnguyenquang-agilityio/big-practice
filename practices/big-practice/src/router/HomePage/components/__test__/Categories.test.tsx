// Libs
import { MemoryRouter } from 'react-router-dom';
import { screen, render, RenderResult } from '@testing-library/react';

// Components
import { Categories } from '../CategoryListContainer';

const mockListCollection = [
  { value: 'phones', label: 'Phone' },
  { value: 'laptops', label: 'Laptop' },
];

describe('Categories component', () => {
  let comp: RenderResult;
  beforeEach(() => {
    comp = render(
      <MemoryRouter>
        <Categories
          categories={mockListCollection}
          selectingItem='smartphones'
        />
      </MemoryRouter>,
    );
  });

  test('Should render correctly', () => {
    expect(comp).toBeTruthy();
  });

  test('Should match snapshot', () => {
    expect(comp).toMatchSnapshot();
  });

  test('Should render correct number of items', async () => {
    const items = await screen.getByTestId('categories');
    expect(items.children).toHaveLength(2);
  });

  test('Should has no underline item in first render', async () => {
    const items = await screen.getByTestId('categories');
    expect(items.getElementsByClassName('font-bold underline').length).toBe(0);
  });
});
