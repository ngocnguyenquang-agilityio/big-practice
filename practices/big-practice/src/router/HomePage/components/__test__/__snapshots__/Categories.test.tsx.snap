// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Categories component Should match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav>
        <h3
          class="hidden text-xs text-neutral-400 md:block"
        >
          Category
        </h3>
        <ul
          class="hidden md:block"
          data-testid="categories"
        >
          <li
            class="mt-2 flex text-white "
          >
            <a
              class="w-full text-sm underline-offset-4 hover:underline hover:text-neutral-100"
              href="/category/phones"
            >
              Phone
            </a>
          </li>
          <li
            class="mt-2 flex text-white "
          >
            <a
              class="w-full text-sm underline-offset-4 hover:underline hover:text-neutral-100"
              href="/category/laptops"
            >
              Laptop
            </a>
          </li>
        </ul>
        <div
          class="md:hidden"
        >
          <div
            class="relative"
          >
            <select
              class="bg-transparent flex w-full items-center justify-between rounded border px-4 py-2 text-sm border-white/30"
            >
              <option
                class="mt-2 flex text-white bg-black"
                value="phones"
              >
                Phone
              </option>
              <option
                class="mt-2 flex text-white bg-black"
                value="laptops"
              >
                Laptop
              </option>
            </select>
          </div>
        </div>
      </nav>
    </div>
  </body>,
  "container": <div>
    <nav>
      <h3
        class="hidden text-xs text-neutral-400 md:block"
      >
        Category
      </h3>
      <ul
        class="hidden md:block"
        data-testid="categories"
      >
        <li
          class="mt-2 flex text-white "
        >
          <a
            class="w-full text-sm underline-offset-4 hover:underline hover:text-neutral-100"
            href="/category/phones"
          >
            Phone
          </a>
        </li>
        <li
          class="mt-2 flex text-white "
        >
          <a
            class="w-full text-sm underline-offset-4 hover:underline hover:text-neutral-100"
            href="/category/laptops"
          >
            Laptop
          </a>
        </li>
      </ul>
      <div
        class="md:hidden"
      >
        <div
          class="relative"
        >
          <select
            class="bg-transparent flex w-full items-center justify-between rounded border px-4 py-2 text-sm border-white/30"
          >
            <option
              class="mt-2 flex text-white bg-black"
              value="phones"
            >
              Phone
            </option>
            <option
              class="mt-2 flex text-white bg-black"
              value="laptops"
            >
              Laptop
            </option>
          </select>
        </div>
      </div>
    </nav>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
